packages/service-library/src/servicelib/fastapi/long_running/_context_manager.py

```diff
@@ -0,0 +1,103 @@
+import asyncio
+from asyncio.log import logger
+from contextlib import asynccontextmanager
+from typing import Any, AsyncIterator, Awaitable, Callable, Optional
+
+from fastapi import FastAPI
+from httpx import AsyncClient
+from pydantic import AnyHttpUrl
+
+from ._client import Client
+from ._errors import TaskClientTimeoutError
+from ._models import TaskId, TaskStatus
+
+
+class _ProgressManager:
```

Comment:
Sorry I really do not see how `tqdm` could be used here, since the aim of this **private** `_ProgressMnager` is to reduce repetitive progress updates incoming from status polling.
It will be the caller's responsibility to display these updates, he can choose to use `tqdm` or maybe send the progress via web sockets to the frontend.